io.openems.edge.batteryinverter.deye/src/io/openems/edge/deye/productionmeter/PvInverterDeye.java

```diff
@@ -0,0 +1,133 @@
+		PDC(Doc.of(OpenemsType.INTEGER) //
+				.unit(Unit.WATT)), //
+		UDC(Doc.of(OpenemsType.INTEGER) //
+				.unit(Unit.VOLT)),
+		YESTERDAY_YIELD(Doc.of(OpenemsType.INTEGER) //
+				.unit(Unit.WATT_HOURS)),
+		MONTHLY_YIELD(Doc.of(OpenemsType.INTEGER) //
+				.unit(Unit.WATT_HOURS)),
+		YEARLY_YIELD(Doc.of(OpenemsType.INTEGER) //
+				.unit(Unit.WATT_HOURS)),
+		TOTAL_YIELD(Doc.of(OpenemsType.INTEGER) //
+				.unit(Unit.CUMULATED_WATT_HOURS)),
+		PAC_CONSUMPTION(Doc.of(OpenemsType.INTEGER) //
+				.unit(Unit.WATT)),
+		YESTERDAY_YIELD_CONS(Doc.of(OpenemsType.INTEGER) //
+				.unit(Unit.WATT_HOURS)),
+		MONTHLY_YIELD_CONS(Doc.of(OpenemsType.INTEGER) //
+				.unit(Unit.WATT_HOURS)),
+		YEARLY_YIELD_CONS(Doc.of(OpenemsType.INTEGER) //
+				.unit(Unit.WATT_HOURS)),
+		TOTAL_YIELD_CONS(Doc.of(OpenemsType.INTEGER) //
+				.unit(Unit.CUMULATED_WATT_HOURS)),
+		TOTAL_POWER(Doc.of(OpenemsType.INTEGER) //
+				.unit(Unit.WATT_HOURS_BY_WATT_PEAK)),
+
+		// PV
+		P_LIMIT_TYPE(Doc.of(PLimitType.values()) //
+				.accessMode(AccessMode.WRITE_ONLY)), //
+		P_LIMIT_PERC(Doc.of(OpenemsType.INTEGER) //
+				.accessMode(AccessMode.READ_WRITE) //
+				.unit(Unit.PERCENT)),
+		P_LIMIT(Doc.of(OpenemsType.INTEGER) //
+				.unit(Unit.KILOWATT)),
+		WATCH_DOG_TAG(Doc.of(OpenemsType.INTEGER) //
+				.accessMode(AccessMode.READ_WRITE)), //
+		PV_LIMIT_FAILED(Doc.of(Level.FAULT) //
+				.text("PV-Limit failed"));
+
+		private final Doc doc;
+
+		private ChannelId(Doc doc) {
+			this.doc = doc;
+		}
+
+		@Override
+		public Doc doc() {
+			return this.doc;
+		}
+
+	}
+
+	/**
+	 * Initializes Channel listeners to calculate the
+	 * {@link ElectricityMeter.ChannelId#ACTIVE_POWER}-Channel value as the sum of
+	 * {@link ElectricityMeter.ChannelId#ACTIVE_POWER_L1},
+	 * {@link ElectricityMeter.ChannelId#ACTIVE_POWER_L2} and
+	 * {@link ElectricityMeter.ChannelId#ACTIVE_POWER_L3}.
+	 *
+	 * @param meter the {@link ElectricityMeter}
+	 */
+	public static void calculateSumActivePowerFromPhases(PvInverterDeye meter) {
+		final Consumer<Value<Integer>> calculate = ignore -> {
+			final Integer generatorPower = meter.getActivePowerGen().getNextValue().get();
+			Integer generatorPowerToSet = 0;
+
+			if (generatorPower > 32768) {
+				generatorPowerToSet = generatorPower - 65536;
+			} else {
+				generatorPowerToSet = generatorPower;
+			}
+
+			meter._setActivePower(TypeUtils.sum(//
+					meter.getActivePowerS1Channel().getNextValue().get(), //
+					meter.getActivePowerS2Channel().getNextValue().get(), //
+					meter.getActivePowerS3Channel().getNextValue().get(), //
+					meter.getActivePowerS4Channel().getNextValue().get(), generatorPowerToSet)); //
+		};
+		meter.getActivePowerS1Channel().onSetNextValue(calculate);
+		meter.getActivePowerS2Channel().onSetNextValue(calculate);
+		meter.getActivePowerS3Channel().onSetNextValue(calculate);
+		meter.getActivePowerS4Channel().onSetNextValue(calculate);
+		meter.getActivePowerGen().onSetNextValue(calculate);
```

Comment:
@Sn0w3y Is this the right way to implement the Gen Port on a inverter?

Reply:
What do you mean with "Gen Port" ?

Reply:
Deye does have a port to plug in a Generator source. Like a Diesel generator. So the values what it generates are now written to active power.

Reply:
I think the right behavior for this case would be to implement a new meter for the generator Port. It is a external source which can produce energy, but also take energy. Maye it would be nice to view the values of the generator port in the UI 
<img width="1081" alt="Bildschirmfoto 2024-04-15 um 09 26 19" src="https://github.com/OpenEMS/openems/assets/18745016/ab1233c0-a94f-42ba-9b01-403675bcfc3b">

https://www.akkudoktor.net/forum/deye-wechselrichter/deye-12k-frage-zum-generator-port-wechsel-der-nutzart/